crt/exit.c

```diff
@@ -0,0 +1,71 @@
+#include <myst/syscallext.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <syscall.h>
+#include <unistd.h>
+
+void do_crt_exit(int code);
+
+/*
+ * If we were cloned with CLONE_VM | CLONE_VFORK we are sharing the process
+ * heap between the parent and the child forked process.
+ * All global variables the CRT uses are shared between the two processes.
+ * If either process shuts down through exit() the CRT will clean up the process
+ * by deleting memory, deallocating locks and various other things. In exit() we
+ * call into the kernel to determine if we are a parent of a fork or a child
+ * fork process. If we are a child fork we skip the cleanup and let the parent
+ * process do that. If we are the parent process of forks we will terminate all
+ * child forks, wait for them to exit, then proceed to cleanup If we are neither
+ * a child forked process or parent of one we cleanup as normal.
+ */
+void exit(int code)
```

Comment:
Is this a strong form of ``exit`` where there is a weak from someplace. If so, can you add a short comment?

Reply:
yes, this is the strong version  that overwrites the CRT weak version